app/file.rb

```diff
@@ -74,8 +74,12 @@ def frontmatter
 			YAML.safe_load(open(path), [Date, Time])
 		end
 
-		def to_directory
-			Tint::Directory.new(site, relative_path)
+		def to_h(_=nil)
+			h = super.merge(mime: mime)
+			if image? && size / 2**20 < 10
+				h = h.merge(encoded_image: Base64.encode64(path.open.read))
```

Comment:
Instead of any of these b64 anywhere in the app, we should basically use something similar to the download route that we have to show previews of images.